src/coreclr/vm/jitinterface.cpp

```diff
@@ -4363,15 +4363,15 @@ static BOOL isMoreSpecificTypeHelper(TypeHandle hnd1, TypeHandle hnd2)
         return FALSE;
     }
 
-    // If we have a mixture of shared and unshared types,
-    // consider the unshared type as more specific.
     BOOL isHnd1CanonSubtype = hnd1.IsCanonicalSubtype();
```

Comment:
This can be moved inside the if check. You only need to compute these values if they are used.

Reply:
Addressed